app/src/main/java/com/spascoding/taskycourse/feature_register_screen/data/remote/AuthenticationApi.kt

```diff
@@ -32,7 +32,7 @@ interface AuthenticationApi {
     ): Response<Void>
 
     @GET("/logout")
-    fun logout(
+    suspend fun logout(
         @Header("Authorization") token: String
-    ): Call<Void>
+    ): Response<Void>
```

Comment:
Technically, you're referencing one feature from another here (agenda uses AuthRepository from the auth feature) which is not allowed, since it doesn't keep features reusable. If you'd try to reuse your agenda feature you'd be forced to also reuse the auth feature with it since it includes functionality agenda needs. 
Therefore, even if it might feel counterintuitive, the logout call would make more sense in the agenda feature here.

Also, I just noticed that your feature is called `register_screen`. Feature != screen however. A feature is a set of screens that belong together, so in the scope of Tasky it would be auth (login + register) and agenda (agenda screen, agenda detail screen, photo detail screen and edit text screen).
When deciding what a feature is, the best thing to ask yourself is "Would it make sense to extract this feature here and reuse in possible lots of other projects without any additional code?" While you could of course reuse just the register screen, it would usually come with a login screen. And you wouldn't be able to really reuse the agenda screen without also reusing its detail screen.